app/models/proxy.rb

```diff
@@ -550,7 +554,17 @@ def create_default_proxy_rule
 
   def deploy
     return true unless deployable?
-    apicast_configuration_driven ? deploy_v2 : deploy_v1
+    # service mesh integration uses production environment directly
+    if service_mesh_integration?
+      deploy_service_mesh_integration
+    else
+      apicast_configuration_driven ? deploy_v2 : deploy_v1
```

Comment:
Just like in https://github.com/3scale/porta/pull/610/files#r257338607 , I think the less nested conditions, the easier, so in this case it would be:
```ruby
def deploy
  return true unless deployable?
  # service mesh integration uses production environment directly
  if service_mesh_integration?
    deploy_service_mesh_integration
  elsif apicast_configuration_driven
    deploy_v2
  else
    deploy_v1
  end
end
```